### PR TITLE
[Media] Migrate WPCom media upload to /wp/v2 endpoints

### LINKED ISSUE #14232: [Product] Use `wp/v2` endpoint for uploading images
### Problem

We currently use `rest/v1.1/sites/\(siteID)/media/new` endpoint to upload images. This doesn't work when XMLRPC is disabled on the site.

### Proposed solution

Start using `wp/v2/sites/\(siteID)/media` endpoint to upload images. This endpoint works even when XMLRPC is disabled. 

Internal -  p1728974476767139/1728139321.219389-slack-C6H8C3G23
Internal - p91TBi-6OX-p2

Android - https://github.com/woocommerce/woocommerce-android/issues/12836

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #14232

## Description

This PR updates the handling of the media operations (upload and fetch) to the user the /wp/v2 endpoints instead of the legacy /v1.1 ones, this is needed to fix upload problems for sites with XMLRPC issues.

Changes
- Remove code which was using v1.1 endpoint for uploading media.
- Call v2 endpoint method for all types of sites, including JCP and Application password login. 
- Update unit tests. 

⚠️  There is a known issue related to loading media from WP media library and updating Product ID to media. https://github.com/woocommerce/woocommerce-ios/issues/14550 We can work on this in a separate PR. 

## Steps to reproduce

**a. No Jetpack case (Application Password)**
1. Have a .org site with no Jetpack, 
1. Install "Disable XML-RPC-API" plugin.
1. In the app, login to site with application password,
1. Go to Products > Add Product > Add Product Image 
1. Try uploading images from your device
1. Save and Publish the product and validate that image uploads work.

**b. Jetpack CP site case**
1. Have a .org site with no Jetpack but with a plugin with Jetpack CP (I use "Blaze Ads" and connect WPCOM account while attempting to create a Blaze campaign from the `wp-admin` page)
1. Install "Disable XML-RPC-API" plugin.
1. Setup the Jetpack CP plugin and connect it to your WordPress.com account
1. In the app, login to site with WordPress.com account,
1. Go to Products > Add Product > Add Product Image 
1. Try uploading images from your device
1. Save and Publish the product and validate that image uploads work.

**c. WordPress.com / full Jetpack site case**
1. Have a .org site with Jetpack plugin
1. Install "Disable XML-RPC-API" plugin.
1. Setup the Jetpack plugin and connect it to your WordPress.com account,
1. In the app, login to site with WordPress.com account,
1. Go to Products > Add Product > Add Product Image 
1. Try uploading images from your device
1. Save and Publish the product and validate that image uploads work.

## Testing information

- I tested that images can uploaded for all the three site types. (JCP, Jetpack, Self-hosted application password login).

## Screenshots

Screenshot 2024-11-27 at 9 56 28 PM

https://github.com/user-attachments/assets/8d5e4a74-f2fb-4816-820a-2f51d324e352

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Networking/Networking/Remote/MediaRemote.swift [hunk 1]
```diff
@@ -18,11 +18,7 @@ public protocol MediaRemoteProtocol {
...
                                            completion: @escaping (Result<[WordPressMedia], Error>) -> Void)
-    func uploadMedia(for siteID: Int64,
-                     productID: Int64,
-                     mediaItems: [UploadableMedia],
-                     completion: @escaping (Result<[Media], Error>) -> Void)
-    func uploadMediaToWordPressSite(siteID: Int64,
+    func uploadMedia(siteID: Int64,
                                     productID: Int64,
                                     mediaItem: UploadableMedia,
                                     completion: @escaping (Result<WordPressMedia, Error>) -> Void)
```
Comment (itsmeichigo) [inline Networking/Networking/Remote/MediaRemote.swift]: Super nit: indentation needs to be updated here.

File: RELEASE-NOTES.txt [hunk 2]
```diff
@@ -5,6 +5,7 @@
 - [*] Jetpack Setup: Fixed an issue where the WordPress.com authentication fails when using a passwordless account that's already connected to Jetpack [https://github.com/woocommerce/woocommerce-ios/pull/14501]
 - [*] Jetpack Setup: Fixed an issue with magic link handling when the app is cold started. [https://github.com/woocommerce/woocommerce-ios/pull/14502]
 - [**] Media Library: On sites logged in with application password, when picking image from WordPress Media Library, all images will now load correctly. [https://github.com/woocommerce/woocommerce-ios/pull/14444]
+- [*] Media: Media upload will work for sites where XML-RPC is disabled. [https://github.com/woocommerce/woocommerce-ios/pull/14537]
```
Comment (itsmeichigo) [inline RELEASE-NOTES.txt]: super nit:

```suggestion
- [*] Media: Media upload will work for sites without XML-RPC. [https://github.com/woocommerce/woocommerce-ios/pull/14537]
```

**Review by itsmeichigo — APPROVED** [review]
Tested with simulator iPhone 16 Pro iOS 18.0 and confirmed that uploading images works for sites (non-JP, JCP, or JP sites) without XMLRPC ✅

File: Networking/Networking/Remote/MediaRemote.swift (hunk 1 above)
Comment (selanthiraiyan) [reply Networking/Networking/Remote/MediaRemote.swift]: Good catch. Done in https://github.com/woocommerce/woocommerce-ios/pull/14537/commits/de07dbdb5e199fa0fe51a922c277dc4bf935956f

**Pull request merged**